src/jaxsim/api/frame.py

```diff
@@ -0,0 +1,196 @@
+
+    Args:
+        model: The model to consider.
+        frame_indices: The indices of the frames.
+
+    Returns:
+        The names of the frames.
+    """
+
+    return tuple(
+        idx_to_name(model=model, frame_index=frame_index)
+        for frame_index in frame_indices
+    )
+
+
+# =========
+# Frame APIs
+# =========
+
+
+@functools.partial(jax.jit, static_argnames=("frame_index",))
+def transform(
+    model: js.model.JaxSimModel,
+    data: js.data.JaxSimModelData,
+    *,
+    frame_index: jtp.IntLike,
+) -> jtp.Matrix:
+    """
+    Compute the SE(3) transform from the world frame to the specified frame.
+
+    Args:
+        model: The model to consider.
+        data: The data of the considered model.
+        frame_index: The index of the frame for which the transform is requested.
+
+    Returns:
+        The 4x4 matrix representing the transform.
+    """
+
+    frame = model.description.get().frames[frame_index - model.number_of_links()]
+    parent_link_index = frame.parent.index
+    W_H_parent = js.link.transform(model=model, data=data, link_index=parent_link_index)
+    parent_H_frame = frame.pose
+
+    return W_H_parent @ parent_H_frame
+
+
+@functools.partial(jax.jit, static_argnames=["frame_index", "output_vel_repr"])
+def jacobian(
+    model: js.model.JaxSimModel,
+    data: js.data.JaxSimModelData,
+    *,
+    frame_index: jtp.IntLike,
+    output_vel_repr: VelRepr | None = None,
+) -> jtp.Matrix:
+    """
+    Compute the free-floating jacobian of the frame.
+
+    Args:
+        model: The model to consider.
+        data: The data of the considered model.
+        frame_index: The index of the frame.
+        output_vel_repr:
+            The output velocity representation of the free-floating jacobian.
+
+    Returns:
+        The 6×(6+n) free-floating jacobian of the frame.
+
+    Note:
+        The input representation of the free-floating jacobian is the active
+        velocity representation.
+    """
+
+    output_vel_repr = (
+        output_vel_repr if output_vel_repr is not None else data.velocity_representation
+    )
+
+    # Get the free-floating jacobian of the parent link in body-fixed output representation
+    L = (
+        model.description.get()
+        .frames[frame_index - model.number_of_links()]
+        .parent.index
+    )
+    L_J_WL = js.link.jacobian(
+        model=model, data=data, link_index=L, output_vel_repr=VelRepr.Body
+    )
+
+    # Adjust the output representation
+    match output_vel_repr:
+        case VelRepr.Inertial:
+            W_H_L = js.link.transform(model=model, data=data, link_index=L)
+            W_X_L = jaxlie.SE3.from_matrix(W_H_L).adjoint()
+            W_J_WL = W_X_L @ L_J_WL
+            O_J_WL_I = W_J_WL
+
+        case VelRepr.Body:
+            W_H_L = js.link.transform(model=model, data=data, link_index=L)
+            W_H_F = transform(model=model, data=data, frame_index=frame_index)
+            F_H_L = jnp.linalg.inv(W_H_F) @ W_H_L
```

Comment:
There is no need to compute and matrix inverse, transforms can be inverted by only transposition and sign inversion (more AD friendly) as follows:

```suggestion
            F_H_L = jaxsim.math.Transform.inverse(W_H_F) @ W_H_L
```

Reply:
Oh right thanks! Fixed in https://github.com/ami-iit/jaxsim/pull/148/commits/a268707c782530b8bb9c70d887102d49401c2ba2